src/ui/docengine.cpp

```diff
@@ -5,8 +5,11 @@
 #include <QTextCodec>
 #include <QTextStream>
 #include <QCoreApplication>
+#include <QPushButton>
+
 #include "include/mainwindow.h"
 #include "include/nqqsettings.h"
+#include "include/Sessions/persistentcache.h"
```

Comment:
is this used?

Reply:
Yes, to get a temporary path to store the file to:

```c++
QString filePath = PersistentCache::createValidCacheName(
                PersistentCache::cacheDirPath(),
                outFileName.fileName() )
            .toLocalFile();
```

Returns something like `/home/user/.config/Notepaddqq/tabCache/filename.12345`. Same method as sessions use.

Reply:
Oh ok I see it now, but still I'm not sure why you store it in the cache instead of in the system tmp dir?

Reply:
Also, does the file get deleted after a successful copy? Or we still need it?

Reply:
I just chose the cache because the code for it already exists and works fine. I can use /tmp/ instead if you want but I don't see much of a reason.

The file gets deleted after the save attempt regardless of whether it was successful or not since it has either done its job or the attempt was cancelled. That happens at the end of `trySudoSave`.

Reply:
Perfect, sorry I'm in a quite distracting environment right now :/ you do the squashmerge?

Reply:
No worries. I'm glad you can find some time here and there at all. Slow progress is better than no progress ;)